airflow/security/kerberos.py

```diff
@@ -59,10 +63,7 @@ def renew_from_kt(principal: str | None, keytab: str, exit_on_fail: bool = True)
     # minutes to give ourselves a large renewal buffer.
     renewal_lifetime = f"{conf.getint('kerberos', 'reinit_frequency')}m"
 
-    cmd_principal = principal or conf.get_mandatory_value("kerberos", "principal").replace(
-        "_HOST", get_hostname()
-    )
-
+    cmd_principal = get_kerberos_principle(principal)
```

Comment:
This PR changes code in both core and in providers and it has separated release cycle.
So if we can extract the `airflow/security/kerberos.py` to a separated PR we will get it merge then rebase this PR

This will make release manager life much easier

Reply:
Sure. I have added another PR here #34936

Reply:
https://github.com/apache/airflow/pull/34936 is merged can you please rebase?

Reply:
Done